lms/djangoapps/discussion_api/api.py

```diff
@@ -714,7 +730,7 @@ def delete_thread(request, thread_id):
 
     """
     cc_thread, context = _get_thread_and_context(request, thread_id)
-    if can_delete(cc_thread, context):
+    if has_permission(cc_thread, context):
```

Comment:
The goal of the original code is to move towards a central model where permissions for requested actions are centralized in permissions.py.  The original `can_delete` method allowed for specifically checking the delete action and not a generic permission.

So either:
1. Revert these changes from `can_delete` to `has_permission`.  
2. Update `has_permission` to take in an action parameter where 'delete_thread', 'delete_comment', 'view_thread', etc can be passed.